### PR TITLE
Twixtools and orientation matrix extraction

### PR DESCRIPTION
**Pull request by mbertrait**
This PR solves #153 by moving from [mapVBVD](https://github.com/wtclarke/pymapvbvd) to [twixtools](https://github.com/pehses/twixtools) to read Siemens Twix scan files and storing the orientation matrix in the `hdr` dict.

### CODE REVIEW

File: src/mrinufft/io/siemens.py [hunk 1]
```diff
@@ -42,47 +44,55 @@ def read_siemens_rawdat(
...
-        "n_coils": int(twixObj.image.NCha),
-        "n_shots": int(twixObj.image.NLin),
-        "n_contrasts": int(twixObj.image.NSet),
-        "n_adc_samples": int(twixObj.image.NCol),
-        "n_slices": int(twixObj.image.NSli),
+        "n_coils": int(twixObj['image'].shape[-2]),
+        "n_shots": int(twixObj['image'].shape[-3]),
+        "n_contrasts": int(twixObj['image'].shape[-4]),
+        "n_adc_samples": int(twixObj['image'].shape[-1]),
+        "n_slices": int(int(twixObj['image'].shape[-5])),
```
Comment (paquiteau) [inline src/mrinufft/io/siemens.py:71]:
```suggestion
k: int(v) for k,v in zip(["n_adc_samples", "n_coils", "n_shots", "n_contrasts", "n_slices"], twixObj["image"].shape[::-1]) 
```

**Review by paquiteau — CHANGES_REQUESTED** [review]
Nice ! I have made a suggestion for more pythonic code. Also you need to run black/ ruff on your code

File: src/mrinufft/io/siemens.py (hunk 1 above)
Comment (mbertrait) [reply src/mrinufft/io/siemens.py:71]: I wanted to stick with the previous code "format" but sure

Comment (paquiteau) [reply src/mrinufft/io/siemens.py:71]: well at least put them in the correct order :D

**Review by chaithyagr — APPROVED** [review]
LGTM, I think Only Pierre's comments makes sense and some code linting. Ideally we need a test, but thats hard as we need to create dummy data. Lets skip testing this module but maybe @mbertrait can you open an issue if its not already present to track the need for a test?

**Comment by chaithyagr** [comment]
A minor comment, if you can it could help to just tell here is twixtools is faster or pymapvbvd. It doesnt have to be an extensive test. Careful though, we dont want bad numbers just because data was in cache.

File: src/mrinufft/io/siemens.py (hunk 1 above)
Comment (mbertrait) [reply src/mrinufft/io/siemens.py:71]: You're right maybe my goal of "sticking with the previous code and only change func calls etc." was at the cost of "less cleaner code". Let's go your way

**Review by chaithyagr — CHANGES_REQUESTED** [review]
Please also change `pymapvbvd` to `twixtools` with the right github link in pyproject.toml
@paquiteau to comment how to specify gitghub link, as twixtools is not released by authors.

**Comment by mbertrait** [comment]
> LGTM, I think Only Pierre's comments makes sense and some code linting. Ideally we need a test, but thats hard as we need to create dummy data. Lets skip testing this module but maybe @mbertrait can you open an issue if its not already present to track the need for a test?

I thought about tests but then as you mentioned we need dummy data so I set them aside for the moment.

**Comment by mbertrait** [comment]
> Please also change `pymapvbvd` to `twixtools` with the right github link in pyproject.toml @paquiteau to comment how to specify gitghub link, as twixtools is not released by authors.

@paquiteau I found a syntax you can find in https://github.com/mind-inria/mri-nufft/pull/157/commits/612cfd9ce23613296376523138e89be10939a3a2 need your expertise on this

**Review by chaithyagr — APPROVED** [review]

**Review by paquiteau — APPROVED** [review]

**Pull request merged**